.github/workflows/kind_upgrade.yml

```diff
@@ -215,11 +215,10 @@ jobs:
   artifact-cleanup:
     name: Delete uploaded images
     needs: [build-antrea-image, from-N-1, from-N-2, compatible-N-1, compatible-N-2]
-    if: ${{ always() }}
+    if: ${{ needs.build-antrea-image.result == 'success' }}
```

Comment:
in my experience, you need something like this:

```
if: ${{ always() && needs.build-antrea-image.result == 'success' }}
```

Otherwise `success() &&` is implicitly added, and the job won't run if any of the jobs listed in `needs` fails.

From the Github docs:

> You can use the following status check functions as expressions in if conditionals. A default status check of success() is applied unless you include one of these functions.

Reply:
done